src/sentry/api/endpoints/project_release_details.py

```diff
@@ -141,6 +143,13 @@ def put(self, request, project, version):
                     datetime=release.date_released,
                 )
 
+            analytics.record(
+                "release.set_commits",
+                user_id=request.user.id if request.user and request.user.id else None,
+                organization_id=project.organization_id,
+                project_ids=[project.id for project in release.projects.all()],
+                user_agent=request.META.get("HTTP_USER_AGENT", ""),
+            )
```

Comment:
Seems like a lot of duplicated code -- what are your thoughts on creating a function that does the analytics?

Reply:
On second thought, do we even need this? The `--local` feature only sends it to the organization releases endpoint iirc.

Reply:
@manuzope refactored to use a mixin